tests/models/animals/test_animal_model.py

```diff
@@ -0,0 +1,80 @@
+"""Test module for model.py (and associated functionality).
+
+This module tests the functionality of model.py, as well as other bits of code that
+define models based on the class defined in model.py
+"""
+
+from contextlib import nullcontext as does_not_raise
+from logging import INFO
+
+import pytest
+from numpy import datetime64, timedelta64
+
+from tests.conftest import log_check
+from virtual_rainforest.models.animals.animal_model import AnimalModel
+
+
+def test_animal_model_initialization(caplog):
+    """Test `AnimalModel` initialization."""
+
+    # Initialize model
+    model = AnimalModel(timedelta64(1, "W"), datetime64("2022-11-01"))
+
+    # In cases where it passes then checks that the object has the right properties
+    assert set(["setup", "spinup", "update", "cleanup"]).issubset(dir(model))
+    assert model.model_name == "animal"
+    assert str(model) == "A animal model instance"
+    assert (
+        repr(model)
+        == "AnimalModel(update_interval = 1 weeks, next_update = 2022-11-08)"
+    )
+
+
+@pytest.mark.parametrize(
+    "config,time_interval,raises,expected_log_entries",
+    [
+        (
+            {},
+            None,
+            pytest.raises(KeyError),
+            (),  # This error isn't handled so doesn't generate logging
```

Comment:
Maybe add a third test where an exception is raised and it is handled, for completeness.